pkg/services/ngalert/notifier/alertmanager_config.go

```diff
@@ -36,57 +38,39 @@ func (moa *MultiOrgAlertmanager) GetAlertmanagerConfiguration(ctx context.Contex
 	if err != nil {
 		return definitions.GettableUserConfig{}, fmt.Errorf("failed to get latest configuration: %w", err)
 	}
-	cfg, err := Load([]byte(query.Result.AlertmanagerConfiguration))
+
+	result, err := moa.gettableUserConfigFromAMConfigString(ctx, org, query.Result.AlertmanagerConfiguration)
 	if err != nil {
-		return definitions.GettableUserConfig{}, fmt.Errorf("failed to unmarshal alertmanager configuration: %w", err)
+		return definitions.GettableUserConfig{}, err
 	}
 
-	result := definitions.GettableUserConfig{
-		TemplateFiles: cfg.TemplateFiles,
-		AlertmanagerConfig: definitions.GettableApiAlertingConfig{
-			Config: cfg.AlertmanagerConfig.Config,
-		},
-	}
+	return result, nil
+}
 
-	for _, recv := range cfg.AlertmanagerConfig.Receivers {
-		receivers := make([]*definitions.GettableGrafanaReceiver, 0, len(recv.PostableGrafanaReceivers.GrafanaManagedReceivers))
-		for _, pr := range recv.PostableGrafanaReceivers.GrafanaManagedReceivers {
-			secureFields := make(map[string]bool, len(pr.SecureSettings))
-			for k := range pr.SecureSettings {
-				decryptedValue, err := moa.Crypto.getDecryptedSecret(pr, k)
-				if err != nil {
-					return definitions.GettableUserConfig{}, fmt.Errorf("failed to decrypt stored secure setting: %w", err)
-				}
-				if decryptedValue == "" {
-					continue
-				}
-				secureFields[k] = true
-			}
-			gr := definitions.GettableGrafanaReceiver{
-				UID:                   pr.UID,
-				Name:                  pr.Name,
-				Type:                  pr.Type,
-				DisableResolveMessage: pr.DisableResolveMessage,
-				Settings:              pr.Settings,
-				SecureFields:          secureFields,
-			}
-			receivers = append(receivers, &gr)
-		}
-		gettableApiReceiver := definitions.GettableApiReceiver{
-			GettableGrafanaReceivers: definitions.GettableGrafanaReceivers{
-				GrafanaManagedReceivers: receivers,
-			},
-		}
-		gettableApiReceiver.Name = recv.Name
-		result.AlertmanagerConfig.Receivers = append(result.AlertmanagerConfig.Receivers, &gettableApiReceiver)
+func (moa *MultiOrgAlertmanager) GetSuccessfullyAppliedAlertmanagerConfigurations(ctx context.Context, org int64, limit int) (definitions.GettableUserConfigs, error) {
```

Comment:
The naming in the PR is way too verbose.

Think about the scope and context and cut the name short. You could use something like `AlertmanagerConfigurations`, which is more [idiomatic golang](https://go.dev/doc/effective_go#Getters) as the words `Get` is not usually used.

This is mostly present in the Grafana codebase because we have people coming from different backgrounds, which are not familiar with golang.

I would highly encourage you to cut all those names short, as it makes it overall very hard to read.

Also, please add comments to the public functions.

Reply:
I may be wrong, but I think that convention applies for getters, i.e. a method that returns an unexported field from a struct. This is not the case here since we're querying a database, not returning something that's already part of `MultiOrgAlertmanager`. In the same file we have method names like `GetAlertmanagerConfiguration` and `GetLatestAlertmanagerConfiguration`, which have the "Get" prefix because they also query the database to get the returned values.

I agree with the name being too verbose, but `AlertmanagerConfigurations` implies that we return **all** AM configs. Maybe something like `GetSuccessfulAlertmanagerConfigurations` would be better: It's shorter and still says what the method does (filtering old AM configs by their `successfully_applied_at` field).

I'll add comments to these functions too :)

Reply:
+1 on `GetSuccessfulAlertmanagerConfigurations`.